ext/AbstractFFTsTestExt.jl

```diff
@@ -71,9 +74,10 @@ function TestUtils.test_plan(P::AbstractFFTs.Plan, x::AbstractArray, x_transform
     end
 end
 
-function TestUtils.test_plan_adjoint(P::AbstractFFTs.Plan, x::AbstractArray; real_plan=false, copy_input=false)
+function TestUtils.test_plan_adjoint(P::AbstractFFTs.Plan, x::AbstractArray;
+                                     real_plan=false, copy_input=false, test_wrappers=true)
     _copy = copy_input ? copy : identity
-    y = rand(eltype(P * _copy(x)), size(P * _copy(x)))
+    y = map(a -> rand(typeof(a)), P * _copy(x)) # generically construct rand array
```

Comment:
Can we use something more idiomatic such as
```suggestion
    P_x = P * _copy(x)
    y = rand!(similar(P_x))
```
(`similar` could be skipped if we are sure that `P_x` is mutable`)

Reply:
I actually had this earlier, and removed it: https://github.com/JuliaMath/AbstractFFTs.jl/pull/118#pullrequestreview-1615665823

Although `map` and just using the eltype is clunkier, it doesn't assume mutability, and fits with the solution we came up with for mapping in the adjoint plans. Also, it avoids a dependency on `Random`, which would have to be depended on by `AbstractFFTs` itself so that it is accessible in the extension. Does that seem reasonable?

Reply:
Oh, I see how `similar` allows us to supprt mutability. I think it's still a little finnicky because `y` is now a slightly different type, and there's also the `Random` dependency. Both should be fine for the FFTW and CUDA use cases, so I'm happy to go with what you think is best.

Reply:
The Random dependency would indeed be annoying. However, `map` could also lead to different types, it doesn't guarantee that `y` is of the same type as `P * x`.

To address your concerns, maybe one could use
```suggestion
    P_x = P * _copy(x)
    z = similar(P_x)
    Tz = eltype(z)
    @inbounds for i in eachindex(z)
        z[i] = rand(Tz)
    end
    y = convert(typeof(P_x), z)
```
`rand!` for generic arrays falls back to this loop anyway (with a `Sampler` which is a bit more efficient in general but does not really matter here IMO): https://github.com/JuliaLang/julia/blob/fb01dd28a5fbabdae6cb5f23c8493db9a377fda3/stdlib/Random/src/Random.jl#L269-L274
And the `convert` should address concerns about slightly different types and avoid e.g. changing from `StaticArrays.SArray` to `StaticArrays.MArray`.

Reply:
I believe the scalar indexing might be problematic for CUDA arrays, at the very least causing warnings.